tools/fragpipe/macros.xml

```diff
@@ -0,0 +1,1206 @@
+    </xml>
+
+    <!--
+      Prepares enviroment variables for FragPipe.
+    -->
+    <token name="@PREPARE_ENVIRONMENT@"><![CDATA[
+        mkdir -p fragpipe_local/lib &&
+        for dir in bin philosopher tools workflows; do
+            ln -s \${CONDA_PREFIX}/share/fragpipe*/\${dir} fragpipe_local/\${dir};
+        done &&
+        for file in \${CONDA_PREFIX}/share/fragpipe*/lib/*; do
+            ln -s \${file} fragpipe_local/lib/\$(basename \${file});
+        done &&
+        cp \${CONDA_PREFIX}/share/fragpipe*/lib/fragpipe-*.jar fragpipe_local/lib/ &&
+        mkdir .cache .config tmp &&
+        export XDG_CONFIG_HOME=\$(pwd)/.config &&
+        export XDG_CACHE_HOME=\$(pwd)/.cache &&
+        export JAVA_OPTS="\$JAVA_OPTS -Djava.io.tmpdir=\$(pwd)/tmp" &&
+        export LD_LIBRARY_PATH="\$LD_LIBRARY_PATH:/usr/local/lib" &&
+    ]]></token>
+
+    <!--
+      Prepares input scan files.
+
+      Scans and moved to a scan directory and linked.
+
+      Annotation files are copied into the scan directory for TMT workflows.
+    -->
+    <token name="@LINK_SAMPLES@"><![CDATA[
+        #import os
+        #set CWD = os.getcwd()
+        CWD=`pwd | sed 's/\//\\\\\//g'` &&
+
+        mkdir -p scans &&
+        #if $wf.workflow_name == "TMT10" or $wf.workflow_name == "TMT11"
+            cp -p '$wf.annotation' 'scans/annotation.txt' &&
+        #end if
+
+        sed "s/^/\$CWD\/scans\//" $manifest > fp.manifest &&
+
+        #for $sf in $inputs
+            #set $i_name = $ln_name($sf)
+            ln -s '$sf' 'scans/${i_name}' &&
+        #end for
+    ]]></token>
+
+    <!--
+      Prepares input files, excluding database.
+    -->
+    <token name="@PREPARE_INPUTS@"><![CDATA[
+        mkdir -p outputs &&
+        @CMD_IMPORTS@
+        @LINK_SAMPLES@
+    ]]></token>
+
+    <!--
+      Prepares database.
+
+      First genericizes unknown sequence headers in the database.
+      Then adds decoys, or decoys and contaminants depending on user selection.
+    -->
+    <token name="@PREPARE_DATABASE@"><![CDATA[
+        '$__tool_directory__/genericize_db.py' $database_name genericized_db.fas &&
+
+        #if str($database_options) != 'none'
+            philosopher workspace --init &&
+            philosopher database --custom genericized_db.fas --prefix rev_
+            #if str($database_options) != 'decoy_and_contam'
+                --contam
+            #end if
+            --contamprefix &&
+            ln -s *-decoys-*.fas final_db.fas &&
+        #else
+            ln -s genericized_db.fas final_db.fas &&
+        #end if
+    ]]></token>
+
+    <!--
+      Prepares outputs.
+
+      Creates log file symlink, makes a new directory for "combined" outputs and creates "concatenated" results.
+    -->
+    <token name="@PREPARE_OUTPUTS@"><![CDATA[
+        cd outputs &&
+        ln -s log_*.txt log.txt &&
```

Comment:
There is only one `log_*.txt` file?

Reply:
Yes, but it is dated. I've moved the wildcard to the `from_work_dir` attribute to be more concise and removed this link.